BeatSaberTwitchIntegration/BeatBot.cs

```diff
@@ -129,18 +130,15 @@ public void Initiate()
                                 var splitInput = inputLine.Split(' ');
                                 if (splitInput[0] == "PING")
                                 {
-                                    _logger.Info("Responded to twitch ping.");
                                     SendMessage("PONG " + splitInput[1]);
-                                    Thread.Sleep(1750);
+                                    continue;
                                 }
 
-                                splitInput = inputLine.Split(':');
-
-                                if (2 >= splitInput.Length) continue;
+                                // ignore server stuff there not privmsg or whisper (remove whisper?)
+                                if (splitInput[2] != "PRIVMSG" && splitInput[2] != "WHISPER") continue;
 
-                                var command = splitInput[2];
-                                OnCommandReceived(command, _currUser);
-                                Thread.Sleep(1750);
```

Comment:
The Thread.Sleep(1750) is a lazy way to account for the twitch rate limiting. It should not be removed without an alternative being implemented.

Reply:
rate limit is with message queue solved